File: leet-code/215. Kth Largest Element in an Array/KthLargestElementInAnArray215_3.java
package com.example;

import java.util.Arrays;
import java.util.Collections;
import java.util.PriorityQueue;
import java.util.stream.IntStream;

public class KthLargestElementInAnArray215_3 {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        Arrays.stream(nums).forEach(pq::offer);
        IntStream.range(1, k).forEach(i -> pq.poll());
        return pq.poll();
    }
}

File: leet-code/215. Kth Largest Element in an Array/KthLargestElementInAnArray215_3.kt
package com.example

import java.util.*

class KthLargestElementInAnArray215_3 {
    fun findKthLargest(nums: IntArray, k: Int): Int {
        val pq = PriorityQueue<Int>(Comparator { o1, o2 -> o2 - o1 })
        pq.addAll(nums.asIterable())
        IntRange(0, k - 2).forEach { _ -> pq.poll() }
        return pq.poll()
    }
}
